### PR TITLE
Restore Teleporter functionality

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Restore Teleporter functionality. Concerning compatibility, https://caniuse.com/url tells us it can be used in any sufficiently recent browser except IE and Opera Mini. If you have a better approach, I'd love to see it.

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: scripts/pi-hole/js/settings-teleporter.js [hunk 1]
```diff
@@ -70,3 +71,27 @@ function importZIP() {
...
+    method: "GET",
+    xhrFields: {
+      responseType: "blob",
+    },
+    success: function (data) {
+      var a = document.createElement("a");
+      // eslint-disable-next-line compat/compat
+      var url = window.URL.createObjectURL(data);
+      a.href = url;
+      a.download = "teleporter.zip";
```
Comment (yubiuser) [inline scripts/pi-hole/js/settings-teleporter.js]: v5 contained a timestamp in the suggested filename. I think this would be a nice addition

Comment (DL6ER) [reply scripts/pi-hole/js/settings-teleporter.js]: Implemented in the latest commit

**Review by yubiuser — COMMENTED** [review]
Nice work. However, while my header is `attachment; filename="pi-hole_nanopi_teleporter_2023-10-12_22-19-38.zip"`, my suggested file name is `_pi-hole_nanopi_teleporter_2023-10-12_22-19-38.zip_`.

See the leading and trainling `_`

**Comment by rdwebdesign** [comment]
This is happening because the **quotes** are part of the string and they are replaced by `_`.

**Comment by rdwebdesign** [comment]
When the button is clicked, the API sends a payload containing the zip file. To download the file, code creates a "fake" link, triggers a Click event and removes the link after the download.

This is how the link looks like:
```html
<a href="blob:http://192.168.0.20…-a302fce919ff" download='"pi-hole_pidev_teleporter_2023-10-16_23-54-13.zip"'>
```
_(The value of `download` parameter is received with double-quotes (`"pi-hole_pidev_teleporter_2023-10-16_23-54-13.zip"`)_.


Each browser deals with these quotes in slightly different ways:
- Firefox just removes them. The file name is correctly used.
- Chrome and Edge replace them with `_`. The file name starts and ends with underscores (`_pi-hole_pidev_teleporter_2023-10-16_23-54-13.zip_`).
- Safari (iOS) changes the file name into `"pi-hole_pidev_teleporter_2023-10-16_23-54-13.zip".zip` (very strange).

I didn't test other browsers.


These are the Response Headers received:
```
cache-control: no-cache, no-store, must-revalidate, private, max-age=0
connection: close
content-disposition: attachment; filename="pi-hole_pidev_teleporter_2023-10-16_23-54-13.zip"
content-length: 16104
content-security-policy: default-src 'self' 'unsafe-inline';
content-type: application/zip
date: Tue, 17 Oct 2023 02:54:13 GMT
expires: 0
pragma: no-cache
referrer-policy: strict-origin-when-cross-origin
x-content-type-options: nosniff
x-frame-options: DENY
x-xss-protection: 0
```

Do we need to send the filename with quotes and remove them on the javascript? Or can we send it without quotes? 

https://github.com/pi-hole/FTL/blob/529d7f7a3013a472b9387256232e560571d77932/src/api/teleporter.c#L38-L40

Can we use this?
```c
        snprintf(pi_hole_extra_headers, sizeof(pi_hole_extra_headers),
                 "Content-Disposition: attachment; filename=%s",
                 filename);
```

**Review by rdwebdesign — APPROVED** [review]

**Review by yubiuser — APPROVED** [review]
The final messages says

```
 Import successful
Processed files:

/etc/pihole/pihole.toml
/etc/pihole/dhcp.leases
/etc/pihole/gravity.db

```

However, it's not gravity.db that have been restored by only the lists, clients, groups,.... We may need to add a hint that users should run `pihole -g`.

**Pull request merged**